quarkus-spring-beans-api/pom.xml

```diff
@@ -0,0 +1,73 @@
+                <artifactId>maven-shade-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <createSourcesJar>true</createSourcesJar>
+                            <artifactSet>
+                                <includes>
+                                    <include>org.springframework:spring-beans</include>
+                                </includes>
+                            </artifactSet>
+                            <filters>
+                                <filter>
+                                    <artifact>org.springframework:spring-beans</artifact>
+                                    <includes>
+                                        <!-- Used in Quarkus -->
+                                        <include>org/springframework/beans/factory/annotation/Autowired**</include>
+                                        <include>org/springframework/beans/factory/annotation/Qualifier**</include>
+                                        <include>org/springframework/beans/factory/annotation/Value**</include>
+                                        <!-- Transitive dependencies -->
+                                        <include>org/springframework/beans/BeansException**</include>
+                                        <include>org/springframework/beans/PropertyAccessException**</include>
+                                        <include>org/springframework/beans/TypeConverter**</include>
+                                        <include>org/springframework/beans/TypeMismatchException**</include>
+                                        <include>org/springframework/beans/factory/ObjectFactory**</include>
+                                        <include>org/springframework/beans/factory/NoSuchBeanDefinitionException**</include>
+                                        <include>org/springframework/beans/factory/NoUniqueBeanDefinitionException**</include>
+                                        <include>org/springframework/beans/factory/annotation/Autowire**</include>
+                                        <include>org/springframework/beans/factory/config/AutowireCapableBeanFactory**</include>
+                                        <include>org/springframework/beans/factory/config/DependencyDescriptor**</include>
+                                        <include>org/springframework/beans/factory/support/AbstractBeanDefinition**</include>
+                                    </includes>
+                                </filter>
+                            </filters>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
```

Comment:
Can you check that the resulting artifact does not force resolving the original artifacts and its dependencies?

Reply:
> LGTM, I made a few comments.
> 
> Now the main questions are:
> 
>     * should these artifacts have dependencies between them (like core depending on bean or whatever, I don't know the original graph)
I don't think so, not sure to understand this question
> 
>     * we need to double-check we don't resolve the original transitive dependencies
I seems ok.